src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -145,32 +155,42 @@ public GetI2CRegister()
+                    if (this.Raw)
+                    {
+                        WriteObject(value);
+                    }
+                    else
+                    {
+                        I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register);
+                        result.Data = value;
+                        WriteObject(result);
+                    }
                 }
                 else
                 {
-                    I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register);
-                    result.Data = value;
-                    WriteObject(result);
+                    byte value = this.Device.device.ReadAddressByte(this.Register);
+                    if (this.Raw)
+                    {
+                        WriteObject(value);
+                    }
+                    else
+                    {
+                        I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register, new byte[1] { value });
+                        WriteObject(result);
+                    }
                 }
             }
-            else
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
             {
-                byte value = this.Device.device.ReadAddressByte(this.Register);
-                if (this.Raw)
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
                 {
-                    WriteObject(value);
-                }
-                else
-                {
-                    I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register, new byte[1] { value });
-                    WriteObject(result);
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
                 }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?

---

Updated